src/main/java/org/folio/services/invoice/InvoiceLineService.java

```diff
@@ -75,12 +80,19 @@ public Future<List<InvoiceLine>> getInvoiceLinesRelatedForOrder(List<String> ord
         .filter(invoiceLine -> orderPoLineIds.contains(invoiceLine.getPoLineId())).collect(toList()));
   }
 
-  public Future<Void> persistInvoiceLines(List<InvoiceLine> lines,  RequestContext requestContext) {
-    var futures = lines.stream()
-      .map(invoiceLine -> persistInvoiceLine(invoiceLine, requestContext))
-      .collect(Collectors.toList());
-    return GenericCompositeFuture.join(futures).mapEmpty();
+  public Future<List<InvoiceLine>> createInvoiceLines(List<InvoiceLine> invoiceLines,  RequestContext requestContext) {
```

Comment:
Hello @damien-git, could you say please can we return invoiceLines creation result (something like InvoiceHelper.createInvoiceAndLines() method) with errors for particular invoice lines in case POST requests fail for some of them? So that we could build such errors for those invoice lines in the CreateInvoiceEventHandler (by INVOICE_LINES_ERRORS_KEY). It will allow to show error only for invoice lines that could not be created, but not for lines that were created.

Reply:
I think an error at this point (when `InvoiceLineService.createInvoiceLines()` is called) means that there is an error for all lines, because other related operations for all lines are not done yet. For instance, the invoice-order relations, the invoice po numbers and the invoice update. Even if an invoice line record is created, the whole operation to create the invoice line (which includes updating the invoice) is not finished. I don't think there is a way around, although I guess we could add code that deletes created lines (if any) when the code fails after the start of the line record creations. Note that this would be an internal error, so it should not happen in normal circumstances.